src/oco_devisen/main.py

```diff
@@ -0,0 +1,127 @@
+# region imports
+from AlgorithmImports import *
+
+# endregion
+
+
+class OcODevisenStrategy(QCAlgorithm):
+
+    def initialize(self):
+        self.set_start_date(2023, 4, 14)
+        self.set_cash(100000)
+        berlin_time_zone_utc_plus_2 = "Europe/Berlin"
+        self.set_time_zone(berlin_time_zone_utc_plus_2)
+        self._eur_usd_symbol = self.add_forex(
+            ticker="EURUSD", resolution=Resolution.MINUTE
+        ).symbol
+        self.orders = {}
+        self.pip_size = 0.00001
+        self.bar_window = RollingWindow[TradeBar](
+            61
+        )  # TODO learn how to use consolidator for this
+
+    def is_place_order_time(self) -> bool:
+        return time(9, 0) > self.time.time() > time(8, 00)
+
+    # TODO check if this works the same way for forex
+    def get_position_size(self, stop_loss_distance):
+        risk_per_trade = self.portfolio.total_portfolio_value * 0.01
+        risk_per_share = stop_loss_distance
+        position_size = risk_per_trade / risk_per_share
+        return int(position_size)
+
+    # TODO fix this for the minute resolution
+    def stop_loss_distance(self, bar):
+        return (bar.high + self.pip_size) - (bar.low - self.pip_size)
+
+    # TODO fix this for the minute resolution
+    def get_take_profit_price_long(self, bar):
+        return bar.close + (self.stop_loss_distance(bar) * 1, 5)  # TODO parameterize
+
+    # TODO fix this for the minute resolution
+    def get_stop_loss_price_long(self, bar):
+        return bar.close - self.stop_loss_distance(bar)
+
+    # TODO fix this for the minute resolution
+    def get_take_profit_price_short(self, bar):
+        return bar.close - (self.stop_loss_distance(bar) * 1, 5)  # TODO parameterize
+
+    def break_out(self, direction: str) -> bool:
+        if direction == "long":
+            return self.bar_window[0].close > max(
+                bar.high for bar in self.bar_window[:-1]
+            )
+        if direction == "short":
+            return self.bar_window[0].close < min(
+                bar.low for bar in self.bar_window[:-1]
+            )
+
+    def on_data(self, data: Slice):
+        if not self._eur_usd_symbol in data.bars:
+            return
+
+        bar = data.bars[self._eur_usd_symbol]
+
+        self.bar_window.add(bar)
+        if not self.portfolio.invested and self.is_place_order_time():
+            if self.break_out(direction="long"):
+                self.market_order(
```

Comment:
this is not the original idea - here we only go in if we see the breakout. 
we should also test the version of the algo where we place stop orders at both ends, and let them both run, essentially hedging the loss of one with the win of the other.